# 01 Data review

            This study uses whole-blood expression from GSE18123 on GPL570 and
            GPL6244, and GSE6575 on GPL570. These are three platform cohorts from
            two studies, not three independent studies. The counts below describe
            assay records, not a verified number of unrelated people.

            See the [pipeline overview](../../README.md) and
            [data and method references](../../data/reference/REFERENCES.md).

In [1]:
from pathlib import Path
import json
from IPython.display import Markdown, Image, display

current = Path.cwd().resolve()
STUDY = next(p for p in (current, *current.parents)
             if (p / "results/published").is_dir())
RESULTS = STUDY / "results/published"

def read_json(relative):
    path = (RESULTS / relative).resolve()
    if not path.is_relative_to(RESULTS.resolve()) or path.suffix != ".json":
        raise ValueError("Only published JSON summaries may be read")
    return json.loads(path.read_text())

def table(headers, rows):
    lines = ["| " + " | ".join(headers) + " |", "| " + " | ".join(["---"] * len(headers)) + " |"]
    lines.extend("| " + " | ".join(map(str, row)) + " |" for row in rows)
    display(Markdown("\n".join(lines)))

print("Review mode: included aggregate results only; no training or participant-table access.")

Review mode: included aggregate results only; no training or participant-table access.


In [2]:
s = read_json("preparation/summary.json")
table(["Measure", "Count"], [["Retained records", s["samples"]],
    ["ASD", s["ASD"]], ["Control", s["Control"]],
    ["Development records", s["train_samples"]], ["Historical test records", s["test_samples"]],
    ["Shared genes", s["common_genes"]], ["QC exclusions", s["quality_excluded_records"]]])
assert s["train_samples"] + s["test_samples"] == s["samples"]

| Measure | Count |
| --- | --- |
| Retained records | 327 |
| ASD | 204 |
| Control | 123 |
| Development records | 261 |
| Historical test records | 66 |
| Shared genes | 17263 |
| QC exclusions | 4 |

## Preparation and evaluation

            Raw arrays were normalized individually with frozen RMA. Unambiguous
            probes were mapped to Entrez gene IDs, multiple probes were combined
            by their median, and only genes shared across platforms were retained.
            Four GPL570 training controls exceeded the recorded GNUSE quality rule.
            GNUSE was unavailable for GPL6244 core output; unavailable is not a pass.

            The 261 development records form five grouped folds. Each fold was
            predicted using a model fitted on the other folds; the same record was
            not both fitted and validated in that outer step. Tuning occurred
            inside the training portion. This procedure was already completed.

            The separate 66-record internal test set was examined in earlier work.
            It was not reopened for the active model and explanation comparisons.
            Unknown relatedness and blood/cohort confounding remain limitations.

            [Reproduction steps](../pipeline/README.md) and
            [preparation summary](../../results/published/preparation/summary.json).